typed/symdiff_test.go

```diff
@@ -0,0 +1,483 @@
+        scalar: numeric
+    - name: string
+      type:
+        scalar: string
+    - name: bool
+      type:
+        scalar: boolean
+    - name: setStr
+      type:
+        list:
+          elementType:
+            scalar: string
+          elementRelationship: associative
+    - name: setBool
+      type:
+        list:
+          elementType:
+            scalar: boolean
+          elementRelationship: associative
+    - name: setNumeric
+      type:
+        list:
+          elementType:
+            scalar: numeric
+          elementRelationship: associative
+`,
+	quads: []symdiffQuad{{
+		`{"numeric":1}`,
+		`{"numeric":3.14159}`,
+		_NS(_P("numeric")),
+		_NS(),
+	}, {
+		`{"numeric":3.14159}`,
+		`{"numeric":1}`,
+		_NS(_P("numeric")),
+		_NS(),
+	}, {
+		`{"string":"aoeu"}`,
+		`{"bool":true}`,
+		_NS(),
+		_NS(_P("bool")),
+	}, {
+		`{"setStr":["a","b","c"]}`,
+		`{"setStr":["a","b"]}`,
+		_NS(),
+		_NS(),
+	}, {
+		`{"setStr":["a","b"]}`,
+		`{"setStr":["a","b","c"]}`,
+		_NS(),
+		_NS(_P("setStr", _SV("c"))),
+	}, {
+		`{"setStr":["a","b","c"]}`,
+		`{"setStr":[]}`,
+		_NS(),
+		_NS(),
+	}, {
+		`{"setStr":[]}`,
+		`{"setStr":["a","b","c"]}`,
+		_NS(),
+		_NS(
+			_P("setStr", _SV("a")),
+			_P("setStr", _SV("b")),
+			_P("setStr", _SV("c")),
+		),
+	}, {
+		`{"setBool":[true]}`,
+		`{"setBool":[false]}`,
+		_NS(),
+		_NS(_P("setBool", _BV(false))),
+	}, {
+		`{"setNumeric":[1,2,3.14159]}`,
+		`{"setNumeric":[1,2,3]}`,
+		_NS(),
+		_NS(_P("setNumeric", _IV(3))),
+	}},
+}, {
+	name:         "associative list",
+	rootTypeName: "myRoot",
+	schema: `types:
+- name: myRoot
+  struct:
+    fields:
+    - name: list
+      type:
+        namedType: myList
+    - name: atomicList
+      type:
+        namedType: mySequence
+- name: myList
+  list:
+    elementType:
+      namedType: myElement
+    elementRelationship: associative
+    keys:
+    - key
+    - id
+- name: mySequence
+  list:
+    elementType:
+      scalar: string
+    elementRelationship: atomic
+- name: myElement
+  struct:
+    fields:
+    - name: key
+      type:
+        scalar: string
+    - name: id
+      type:
+        scalar: numeric
+    - name: value
+      type:
+        namedType: myValue
+    - name: bv
+      type:
+        scalar: boolean
+    - name: nv
+      type:
+        scalar: numeric
+- name: myValue
+  map:
+    elementType:
+      scalar: string
+`,
+	quads: []symdiffQuad{{
+		`{"list":[{"key":"a","id":1,"value":{"a":"a"}}]}`,
+		`{"list":[{"key":"a","id":1,"value":{"a":"a"}}]}`,
+		_NS(),
+		_NS(),
+	}, {
+		`{"list":[{"key":"a","id":1,"value":{"a":"a"}}]}`,
+		`{"list":[{"key":"a","id":2,"value":{"a":"a"}}]}`,
+		_NS(),
+		_NS(
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2))),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "key"),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "id"),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "value"),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "value", "a"),
+		),
+	}, {
```

Comment:
What happens if I change the value within an associative list? I want to see that you own the field without owning the list item.

Reply:
test added